x-pack/plugins/observability_solution/logs_explorer/public/customizations/custom_control_column.tsx

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { ComponentClass } from 'react';
+import {
+  OPEN_DETAILS,
+  SELECT_ROW,
+  type ControlColumnsProps,
+  DataTableRowControl,
+} from '@kbn/unified-data-table';
+import { EuiButtonIcon, EuiDataGridCellValueElementProps, EuiToolTip } from '@elastic/eui';
+import type { DataTableRecord } from '@kbn/discover-utils/src/types';
+import { useActor } from '@xstate/react';
+import { LogsExplorerControllerStateService } from '../state_machines/logs_explorer_controller';
+import {
+  malformedDocButtonLabelWhenNotPresent,
+  malformedDocButtonLabelWhenPresent,
+  stacktraceAvailableControlButton,
+  stacktraceNotAvailableControlButton,
+} from '../components/common/translations';
+import * as constants from '../../common/constants';
+import { getStacktraceFields } from '../utils/get_stack_trace';
+import { LogDocument } from '../../common/document';
+import { ActionsColumnTooltip } from '../components/virtual_columns/column_tooltips/actions_column_tooltip';
+
+const ConnectedMalformedDocs = ({
+  rowIndex,
+  service,
+}: {
+  rowIndex: number;
+  service: LogsExplorerControllerStateService;
+}) => {
+  const [state] = useActor(service);
+
+  if (state.matches('initialized') && state.context.rows) {
+    const row = state.context.rows[rowIndex];
+    return <MalformedDocs row={row} rowIndex={rowIndex} />;
+  }
+
+  return null;
+};
+
+const ConnectedStacktraceDocs = ({
+  rowIndex,
+  service,
+}: {
+  rowIndex: number;
+  service: LogsExplorerControllerStateService;
+}) => {
+  const [state] = useActor(service);
+
+  if (state.matches('initialized') && state.context.rows) {
+    const row = state.context.rows[rowIndex];
+    return <Stacktrace row={row} rowIndex={rowIndex} />;
+  }
+
+  return null;
+};
+
+const MalformedDocs = ({ row, rowIndex }: { row: DataTableRecord; rowIndex: number }) => {
+  const isMalformedDocumentExists = !!row.raw[constants.MALFORMED_DOCS_FIELD];
```

Comment:
With an empty array this would also evaluate to `true`. Would something like

```suggestion
  const isMalformedDocument = (row.raw[constants.MALFORMED_DOCS_FIELD]?.length ?? 0) > 0;
```

be more robust?

Reply:
@weltenwort So the `constants.MALFORMED_DOCS_FIELD = 'ignored_field_values'`

This property is either present inside `row.raw` in case of malformed docs or not. Hence the double negation check should do the trick.

<img width="485" alt="image" src="https://github.com/elastic/kibana/assets/7416358/c5aefc13-f048-43a9-a226-3da4fafa5445">

Reply:
Oh, I thought `constants.MALFORMED_DOCS_FIELD` was `_ignored`, which holds an array of the fields. :facepalm: nevermind

Reply:
What prompted me to propose is this: When looking at the component and its API in isolation, I would expect a row object of `{"ignored_field_values": {}}` to behave identical to `{}`. And `{"_ignored": []}` analogously.